### ISSUE TITLE
Cannot printf ulong on ARM (cortex-A9)

### ISSUE THREAD

**Issue by jimmysitu** [pocl/pocl#432]
Hi, All
 I am writting a OpenCL program with pocl on Parallella, a zynq(cortex-A9) base device.
And I found that pocl cannot pintf ulong correctly
Here is my code of kernel
```
__kernelvoid search(__global unsigned char* block, volatile __global uint* output) {
  ulong ulongTest = 0xdeadbeefdeadbeef;
  printf("[OCL_KNL]: gid = 0x%08x\n", (uint)get_global_id(0));
  printf("[OCL_KNL]: ulongTest = 0x%08lx\n", ulongTest);
}
```
Output

> [OCL_KNL]: gid = 0x00000000
>[OCL_KNL]: ulongTest = 0xb13428a6b1e7a3a2
>[OCL_KNL]: gid = 0x00000001
>[OCL_KNL]: ulongTest = 0xb13428a6b1e7a3a2
>.......

I am using llvm 3.6 with pocl 0.12(commit aa553e)
Can anyone help?
Thanks.

**Comment by pjaaskel** [comment]
Can you reproduce with pocl master and LLVM 3.9 or LLVM 4.0?

**Comment by jimmysitu** [comment]
Hi, pjaaskel
Yes， I can try pocl master and LLVM 3.9 on my device
But I am using ubuntu 15.04 on ARMv7a, and I do not find any apt package fit for that.
Could you show me how to install llvm 3.9 on my device?
Thanks.

**Comment by kraiskil** [comment]
Note: LLVM 3.9 is broken on ARM. See issue #409

**Comment by pjaaskel** [comment]
Does this reproduce with LLVM 5.0 and pocl master?

**Comment by franz** [comment]
This is fixed in PR #640
